application/apps/indexer/parsers/src/someip.rs

```diff
@@ -337,6 +348,250 @@ impl Display for SomeipLogMessage {
+
+            loop {
+                if cancel.is_cancelled() {
+                    break;
+                }
+                match reader.next() {
+                    Ok((offset, block)) => {
+                        match block {
+                            PcapBlockOwned::NG(pcap_parser::Block::EnhancedPacket(ref epb)) => {
+                                if let Err(e) = read_someip_statistic_from_pcapng_block(
+                                    epb.data,
+                                    &mut services,
+                                    &mut messages,
+                                ) {
+                                    error = Some(e.to_string());
+                                    break;
+                                }
+                            }
+                            PcapBlockOwned::NG(pcap_parser::Block::SimplePacket(ref spb)) => {
+                                if let Err(e) = read_someip_statistic_from_pcapng_block(
+                                    spb.data,
+                                    &mut services,
+                                    &mut messages,
+                                ) {
+                                    error = Some(e.to_string());
+                                    break;
+                                }
+                            }
+                            _ => {
+                                // skipped
+                            }
+                        }
+                        reader.consume(offset);
+                    }
+                    Err(PcapError::Eof) => {
+                        break;
+                    }
+                    Err(PcapError::Incomplete) => {
+                        reader.refill().expect("pcapng refill failed");
+                        // continue;
+                    }
+                    Err(e) => {
+                        let msg = e.to_string();
+                        error!("pcapng at offset {} : {}", reader.consumed(), msg);
+                        error = Some(msg);
+                        break;
+                    }
+                }
+            }
+        }
+        Err(e) => {
+            error = Some(e.to_string());
+        }
+    }
+
+    if let Some(err) = error {
+        return Err(Error::Parse(err));
+    }
+
+    let result = SomeipStatistic {
+        services: map_statistic(&services),
+        messages: map_statistic(&messages),
+    };
+
+    Ok(result)
+}
+
+type StatisticItemMap = FxHashMap<usize, (usize, StatisticDetailMap)>;
+type StatisticDetailMap = FxHashMap<usize, usize>;
+
+fn add_statistic(item_map: &mut StatisticItemMap, item_id: usize, detail_id: usize) {
+    if let Some((item_value, detail_map)) = item_map.get_mut(&item_id) {
+        *item_value += 1;
+        if let Some(detail_value) = detail_map.get_mut(&detail_id) {
+            *detail_value += 1;
+        } else {
+            detail_map.insert(detail_id, 1);
+        }
```

Comment:
this is usually done with the `entry` function (https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry)
if possible use that here as well